src/main/java/seedu/contax/model/AddressBook.java

```diff
@@ -115,14 +117,74 @@ public boolean hasTag(Tag tag) {
         return tags.contains(tag);
     }
 
+    /**
+     * Adds the given tag into the address book.
+     * Tag must not already exist in the address book.
+     */
     public void addTag(Tag tag) {
         tags.add(tag);
     }
 
+    /**
+     * Deletes the given tag from the address book.
+     * The supplied tag must already exist in the address book.
+     */
     public void removeTag(Tag tagToDelete) {
+        removePersonsWithTag(tagToDelete);
         tags.remove(tagToDelete);
     }
 
+    /**
+     * Replaces the given {@code target} with {@code editedTag}.
+     * {@code target} must exist in the address book.
+     * {@code editedTag} must not be the same as another existing tag in the address book.
+     */
+    public void setTag(Tag target, Tag editedTag) {
+        tags.setTag(target, editedTag);
+        setPersonsWithTag(target, editedTag);
+    }
+
+    /**
+     * Disassociates the specified tag from all persons in the address book.
+     */
+    private void removePersonsWithTag(Tag tagToDelete) {
+        requireNonNull(tagToDelete);
+        List<Person> persons = getPersonList().filtered(person -> person.getTags().contains(tagToDelete));
```

Comment:
I think you should be consistent with what you coded below in `setPersonsWithTag` (I think the implementation below is neater).
```suggestion
        List<Person> persons = getPersonList().filtered(person -> person.hasTag(tagToDelete));
```

Reply:
Fixed in 7ff3cf68